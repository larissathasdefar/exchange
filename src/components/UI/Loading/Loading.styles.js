import styled from 'styled-components'

export const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: ${({ fullHeight }) => (fullHeight ? 'calc(100vh - 78px)' : '90px')};
  min-width: 140px;
`
