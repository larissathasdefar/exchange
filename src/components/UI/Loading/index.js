import React from 'react'
import CircularProgress from '@material-ui/core/CircularProgress'
import { Container } from './Loading.styles'

const Loading = props => {
  return (
    <Container {...props}>
      <CircularProgress />
    </Container>
  )
}

export default Loading
