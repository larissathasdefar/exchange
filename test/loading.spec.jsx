import React from 'react'
import { shallow } from 'enzyme'

import Loading from 'ui/Loading'

describe('Loading', () => {
  it('Contains circular progress', () => {
    const component = shallow(<Loading />)
    expect(component.html()).toContain('MuiCircularProgress')
  })

  it('Snapshot', () => {
    const component = shallow(<Loading />)
    expect(component.debug()).toMatchSnapshot()
  })
})