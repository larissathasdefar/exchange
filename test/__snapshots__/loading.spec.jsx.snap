// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Snapshot 1`] = `
"<styled.div>
  <WithStyles(CircularProgress) />
</styled.div>"
`;
